packages/contracts-bedrock/test/L2/L2StandardBridgeInterop.t.sol

```diff
@@ -0,0 +1,369 @@
+        vm.expectRevert(InvalidLegacyERC20Address.selector);
+
+        // Act
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
+
+    /// @notice Test that the `convert` function with an invalid superchain ERC20 address reverts
+    function testFuzz_convert_invalidSuperchainERC20Address_reverts(
+        address _from,
+        address _to,
+        uint256 _amount,
+        address _remoteToken
+    )
+        public
+    {
+        // Assume
+        vm.assume(_remoteToken != address(0));
+
+        // Arrange
+        _setUpSuperToLegacy(_from, _to);
+
+        // Mock the legacy factory to return `_remoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _to, _remoteToken);
+
+        // Mock the superchain factory to return address(0)
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _from, address(0));
+
+        // Expect the revert with `InvalidSuperchainERC20Address` selector
+        vm.expectRevert(InvalidSuperchainERC20Address.selector);
+
+        // Act
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
+
+    /// @notice Test that the `convert` function with different remote tokens reverts
+    function testFuzz_convert_differentRemoteAddresses_reverts(
+        address _from,
+        address _to,
+        uint256 _amount,
+        address _fromRemoteToken,
+        address _toRemoteToken
+    )
+        public
+    {
+        // Assume
+        vm.assume(_fromRemoteToken != address(0));
+        vm.assume(_toRemoteToken != address(0));
+        vm.assume(_fromRemoteToken != _toRemoteToken);
+
+        // Arrange
+        _setUpSuperToLegacy(_from, _to);
+
+        // Mock the legacy factory to return `_fromRemoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _to, _fromRemoteToken);
+
+        // Mock the superchain factory to return `_toRemoteToken`
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _from, _toRemoteToken);
+
+        // Expect the revert with `InvalidTokenPair` selector
+        vm.expectRevert(InvalidTokenPair.selector);
+
+        // Act
+        l2StandardBridge.convert(_from, _to, _amount);
+    }
+
+    /// @notice Test that the `convert` function succeeds
+    function testFuzz_convert_succeeds(
+        address _caller,
+        address _from,
+        address _to,
+        uint256 _amount,
+        address _remoteToken
+    )
+        public
+    {
+        // Assume
+        vm.assume(_remoteToken != address(0));
+
+        // Arrange
+        _setUpSuperToLegacy(_from, _to);
+
+        // Mock the legacy and superchain factory to return `_remoteToken`
+        _mockDeployments(address(l2OptimismMintableERC20Factory), _to, _remoteToken);
+        _mockDeployments(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY, _from, _remoteToken);
+
+        // Expect the `Converted` event to be emitted
+        vm.expectEmit(true, true, true, true, address(l2StandardBridge));
```

Comment:
nit: `vm.expectEmit(bool,bool,bool,bool,address)` where all are true is the same as `vm.expectEmit(address)`